src/vs/workbench/browser/parts/quickopen/quickOpenController.ts

```diff
@@ -479,6 +483,21 @@ export class QuickOpenController extends WorkbenchComponent implements IQuickOpe
 
 		this.previousValue = prefix;
 
+		// Track active editor before navigation
```

Comment:
This is debt: we have two different quick open widgets for the picker and for the quick open main widget (the one that allows to do most things). I am actually not so sure anymore we would want this functionality for the picker so maybe we remove the code that handles the cancellation from the picker and only run it from the quick open widget.